Race condition can lead to build failures on MacOSX

On MacOSX 10.12 Sierra, if multiple Gradle projects using this plugin are built in parallel, a race condition can appear where the Praat binary cannot be extracted and the build fails:
```
    > Task :praat FAILED
    hdiutil: attach failed - Resource busy

    FAILURE: Build failed with an exception.

    * What went wrong:
    Execution failed for task ':praat'.
    > Process 'command 'hdiutil'' finished with non-zero exit value 1

    * Try:
    Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

    * Get more help at https://help.gradle.org

    BUILD FAILED in 53s
```

Also, on OSX 10.14 Mojave, this can happen:
```
    > Task :praat
    Checksumming Protective Master Boot Record (MBR: 0)…
    Protective Master Boot Record (MBR: : checksum failed with error 9.
    calculated CRC32 $00000000, expected   CRC32 $708B4783
    hdiutil: attach failed - Bad file descriptor

    > Task :praat FAILED

    FAILURE: Build failed with an exception.

    * What went wrong:
    Execution failed for task ':praat'.
    > Process 'command 'hdiutil'' finished with non-zero exit value 1

    * Try:
    Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

    * Get more help at https://help.gradle.org

    BUILD FAILED in 41s
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,10 @@ Gradle Praat Wrapper Plugin
 - Update provided Praat to v6.0.30
 - **Note:** running scripts now requires `--run` argument
 
+### Fixed
+
+- Race condition when running `praat` task in parallel on OSX
+
 [v0.5] - 2017-11-20
 -------------------
 

--- a/src/main/groovy/org/m2ci/msp/praat/PraatWrapperPlugin.groovy
+++ b/src/main/groovy/org/m2ci/msp/praat/PraatWrapperPlugin.groovy
@@ -48,8 +48,12 @@ class PraatWrapperPlugin implements Plugin<Project> {
             ext.binary = project.file("$destinationDir/$binary")
             ext.version = praatVersion
             filesMatching('*.dmg') { dmgFileDetails ->
+                project.copy {
+                    from dmgFileDetails.file
+                    into temporaryDir
+                }
                 project.exec {
-                    commandLine 'hdiutil', 'attach', '-mountpoint', temporaryDir, dmgFileDetails.file
+                    commandLine 'hdiutil', 'attach', '-mountpoint', temporaryDir, "$temporaryDir/$dmgFileDetails.file.name"
                 }
                 project.copy {
                     from "$temporaryDir/Praat.app/Contents/MacOS"

--- a/src/test/groovy/org/m2ci/msp/praat/PraatWrapperConcurrentTest.groovy
+++ b/src/test/groovy/org/m2ci/msp/praat/PraatWrapperConcurrentTest.groovy
@@ -1,0 +1,39 @@
+package org.m2ci.msp.praat
+
+import org.gradle.testkit.runner.GradleRunner
+import org.testng.annotations.BeforeClass
+import org.testng.annotations.Test
+
+import static org.gradle.testkit.runner.TaskOutcome.SUCCESS
+import static org.gradle.testkit.runner.TaskOutcome.UP_TO_DATE
+
+class PraatWrapperConcurrentTest {
+
+    GradleRunner gradle
+
+    @BeforeClass
+    void setUp() {
+        def rootProjectDir = File.createTempDir()
+        gradle = GradleRunner.create().withProjectDir(rootProjectDir).withPluginClasspath()
+        new File(rootProjectDir, 'settings.gradle').withWriter { settings ->
+            (1..6).each { p ->
+                def projectDir = new File(rootProjectDir, "proj_$p")
+                settings.println "include '$projectDir.name'"
+                projectDir.mkdirs()
+                new File(projectDir, 'build.gradle').withWriter {
+                    it << this.class.getResourceAsStream('build.gradle')
+                }
+            }
+        }
+        gradle = GradleRunner.create().withProjectDir(rootProjectDir).withPluginClasspath()
+    }
+
+    @Test
+    void testConcurrentExtraction() {
+        def result = gradle.withArguments('testPraat', '--parallel').build()
+        println result.output
+        result.tasks.each { task ->
+            assert task.outcome in [SUCCESS, UP_TO_DATE]
+        }
+    }
+}

